dev/io.openliberty.microprofile.health.4.0.internal/src/io/openliberty/microprofile/health40/services/impl/HealthCheck40CDIBeanInvokerImpl.java

```diff
@@ -150,7 +150,8 @@ private Set<Object> getHealthCheckBeans(BeanManager beanManager, Annotation hcQu
         return healthCheckBeans;
     }
 
-    private BeanManager getBeanManager(String appName, String moduleName) {
+    private synchronized BeanManager getBeanManager(String appName, String moduleName) {
```

Comment:
Would this cause any performance issues for the normal mpHealth functionality, every time we query the endpoints simultaneously?

Reply:
Possibly if that happens. But the bean manager list is small enough that this should return very quickly. The issue is during initial app deployment, where this the map can be modified while being retrieved resulting in a concurrent modification exception.

Reply:
We did see some cases in the past, where an EAR contained at least 10 different WARs/modules, so the bean list might get bigger, in that case. Would be good to get that scenario tested, during our performance testing.